### PR TITLE
document serverConfig.ts

### PR DESCRIPTION
**Pull request by niklasfi**
add some more documentation as to the origins of parameters in serverConfig-example.ts

### CODE REVIEW

File: src/serverConfig-example.ts [hunk 1]
```diff
@@ -3,23 +3,33 @@
...
+  // p2p: {
+  //    enabled: true,
+  //    preferH264: true,
+  //    disableH264: true,
   //    useStunTurn: true,
-  // }, 
+  // },
   // useStunTurn: true,
-  bosh: `https://meet.jit.si/http-bind?room=chatmosphere1234`, // ! if you make your own please omit the "?room=chatmosphere1234" part
+  //bosh: `https://meet.jit.si/http-bind?room=chatmosphere1234`, // ! if you make your own please omit the "?room=chatmosphere1234" part
```
Comment (trilader) [inline src/serverConfig-example.ts]: Why is this now commented out (which is a change from the default)?

File: src/serverConfig-example.ts [hunk 2]
```diff
@@ -3,23 +3,33 @@
...
+    // curl -s https://$JITSI_MEET_WEB_DOMAIN/index.html | grep config.hosts.domain
+    domain: 'meet.jit.si',
+    // curl -s https://$JITSI_MEET_WEB_DOMAIN/index.html | grep config.hosts.muc
+    muc: 'conference.meet.jit.si',
+    // curl -s https://$JITSI_MEET_WEB_DOMAIN/index.html | grep config.hosts.anonymousdomain
+    anonymousdomain: 'guest.meet.jit.si',
+    // curl -s https://$JITSI_MEET_WEB_DOMAIN/index.html | grep config.hosts.authdomain
+    authdomain: 'meet.jit.si',
+  },
+  // propbably https://$JITSI_MEET_WEB_DOMAIN/http-pre-bind
```
Comment (trilader) [inline src/serverConfig-example.ts]: s/propbably/probably/

File: src/serverConfig-example.ts [hunk 3]
```diff
@@ -3,23 +3,33 @@
...
 // Just duplicate the file and name it serverConfig.ts to adjust to your likings
 
+
+// you can derive the following options by looking at the /index.html file of your jitsi-meet-web instance (i.e. the server, where you can create new sessions)
+
 export const connectionOptions = {
   hosts: {
-     domain: 'meet.jit.si',
-     muc: 'conference.meet.jit.si', 
-     focus: 'focus.meet.jit.si',
```
Comment (trilader) [inline src/serverConfig-example.ts:10]: Where did the focus domain go?

**Review by trilader — COMMENTED** [review]
Looks good in general except there are lots of (I think maybe unnecessary) white-space changes

File: src/serverConfig-example.ts (hunk 3 above)
Comment (niklasfi) [reply src/serverConfig-example.ts:10]: don't really know. question is: what is it needed for? apparently, my setup works without one…

File: src/serverConfig-example.ts (hunk 2 above)
Comment (niklasfi) [reply src/serverConfig-example.ts]: fixed in f6a40c2

File: src/serverConfig-example.ts (hunk 1 above)
Comment (niklasfi) [reply src/serverConfig-example.ts]: as above: not sure what it is needed for / whether it is needed.

**Comment by niklasfi** [comment]
> Looks good in general except there are lots of (I think maybe unnecessary) white-space changes

yeah, whitespace: I trimmed quite a bit of trailing whitespace. Do you want it back?

File: src/serverConfig-example.ts (hunk 1 above)
Comment (trilader) [reply src/serverConfig-example.ts]: It's the URL used to contact the underlying Prosody XMPP server via [BOSH](https://xmpp.org/extensions/xep-0124.html). You don't explicitly need it as `/http-bind` is the default in the [library wrapper](https://github.com/jitsi/lib-jitsi-meet/blob/master/modules/xmpp/xmpp.js) which gets passed the options object. Also it looks like it was renamed to `serviceURL` so the value in this config files doesn't take effect but I'm not sure on that.

File: src/serverConfig-example.ts (hunk 3 above)
Comment (trilader) [reply src/serverConfig-example.ts:10]: This allows you to explicitly specify the focus component to use (part of the XMPP stuff) it defaults to the value of `hosts.domain` if not set.

**Comment by trilader** [comment]
> > Looks good in general except there are lots of (I think maybe unnecessary) white-space changes
> 
> yeah, whitespace: I trimmed quite a bit of trailing whitespace. Do you want it back?

I don't care about the trailing whitespace but I think the changes in the leading whitespace create a bigger diff than necessary.
_edit:_ I don't have a problem with fixing the indentation (if I saw that correctly it was 5 spaces) but maybe in a separate commit so the one with the actual changes is more clear?

**Comment by textshell** [comment]
Da das eine nicht funktionale Änderung ist, macht das mehr Sinn zu schauen ob upstream das direkt nimmt?

File: src/serverConfig-example.ts (hunk 3 above)
Comment (niklasfi) [reply src/serverConfig-example.ts:10]: tried to address this in 21e3cf9.

File: src/serverConfig-example.ts [hunk 4]
```diff
@@ -16,20 +15,26 @@ export const connectionOptions = {
...
+  //    disableH264: true, 
   //    useStunTurn: true,
-  // },
+  // }, 
   // useStunTurn: true,
-  //bosh: `https://meet.jit.si/http-bind?room=chatmosphere1234`, // ! if you make your own please omit the "?room=chatmosphere1234" part
+
+  // (optional) bosh can be set to the url of the Prosody XMPP server to be contacted via BOSH.
+  // if you don't set this, it will default to /http-bind
+  bosh: `https://meet.jit.si/http-bind?room=chatmosphere1234`, // ! if you make your own please omit the "?room=chatmosphere1234" part
```
Comment (niklasfi) [inline src/serverConfig-example.ts:34]: @trilader bosh added back in for you

File: src/serverConfig-example.ts [hunk 5]
```diff
@@ -16,20 +15,26 @@ export const connectionOptions = {
     anonymousdomain: 'guest.meet.jit.si',
     // curl -s https://$JITSI_MEET_WEB_DOMAIN/index.html | grep config.hosts.authdomain
     authdomain: 'meet.jit.si',
-  },
+    // (optional) specify the focus component to use (part of the XMPP stuff). defaults to the value of hosts.domain if not set.
+    focus: 'focus.meet.jit.si',
```
Comment (niklasfi) [inline src/serverConfig-example.ts:19]: focus domain added

**Comment by niklasfi** [comment]
> Da das eine nicht funktionale Änderung ist, macht das mehr Sinn zu schauen ob upstream das direkt nimmt?

evtl. ja, da ich aber nicht weiß, wie aktiv upstream ist, würde ich vorschlagen, dass wir es zuerst hier mergen, und ich dann noch mal einen pr gegen upstream mache.

**Comment by trilader** [comment]
Sieht jetzt gut aus. Kannst du das einmal squashen? Ich denke extra commits für die Fixes sind nicht nötig. GH kann das prinzipiell auch direkt im Webinterface aber dann ist der commit von mir wenn ich das richtig sehe.

**Comment by niklasfi** [comment]
@trilader squashed into c7dcc54

**Comment by trilader** [comment]
Thanks

**Pull request merged**